apps/espnnews/espn_news.star

```diff
@@ -0,0 +1,139 @@
+"""
+Applet: ESPN News
+Summary: Get top headlines from ESPN
+Description: Displays the top three headlines from the "Top Headlines" section on ESPN or a specific user-selected sport.
+Author: rs7q5 (RIS)
+"""
+#espn_news.star
+#Created 20211231 RIS
+#Last Modified 20220117 RIS
+
+load("render.star", "render")
+load("http.star", "http")
+load("cache.star", "cache")
+load("schema.star", "schema")
+
+#this list are any of the sports that have a "Top headlines" section and can be done with the following base ESPN_URL
+ESPN_URL = "https://www.espn.com/"
+ESPN_SPORTS_LIST = {
+    "All": ["All", ""],  #default
+    "NFL": ["NFL", "nfl"],
+    "NBA": ["NBA", "nba"],
+    "NHL": ["NHL", "nhl"],
+    "Soccer": ["SOCC", "soccer"],
+    "Golf": ["Golf", "golf"],
+    "NCAAF": ["NCAAF", "college-football"],
+    "College Sports": ["COL.", "college-sports"],
+    "F1": ["F1", "f1"],
+    "MLB": ["MLB", "mlb"],
+    "MMA": ["MMA", "mma"],
+    "NASCAR": ["NASC", "racing/nascar"],
+    "NCAAM": ["NCAAM", "mens-college-basketball"],
+    "NCAAW": ["NCAAW", "womens-college-basketball"],
+    "Olympic Sports": ["OLY", "olympics"],
+    "Racing": ["RCNG", "racing"],
+    "Tennis": ["TENNS", "tennis"],
+    "WNBA": ["WNBA", "wnba"],
+}
+
+def main(config):
+    sport = config.get("sport") or "All"
+    sport_txt, sport_ext = ESPN_SPORTS_LIST.get(sport)
+
+    #create full URL
+    ESPN_API_URL = ESPN_URL + sport_ext
+    if sport != "All":
+        ESPN_API_URL += "/?xhr=1"
+    else:
+        ESPN_API_URL += "?xhr=1"
+
+    font = "CG-pixel-4x5-mono"  #set font
+
+    #check for cached data
+    title_cached = cache.get("title_rate1"), cache.get("title_rate2"), cache.get("title_rate3")
+
+    if all(title_cached) != False:  #if any are None then all(title_cached)==False
+        print("Hit! Displaying cached data.")
+        title = title_cached
+    else:
+        print("Miss! Calling ESPN data.")
+        rep = http.get(ESPN_API_URL)
+        if rep.status_code != 200:
+            fail("ESPN request failed with status %d", rep.status_code)
```

Comment:
We may have to reconsider this `fail()`, but it's fine for now. We're trying to figure out better guidance here - but I think we may land on apps should never `fail()` simply because it will be unsustainable to ensure hundreds of APIs are online and working for all apps.